### PR TITLE
datasource/launch_template: Add new datasource

### LINKED ISSUE #5272: New datasource request: aws_launch_template
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Datasource that returns AWS launch template.

### New Resource

* aws_launch_template

### Potential Terraform Configuration

Similar to aws_launch_configuration: https://terraform.io/docs/providers/aws/d/launch_configuration.html

### PR DESCRIPTION
**Pull request by WhileLoop**
Fixes #5272

Changes proposed in this pull request:

* Ad new aws_launch_template datasource
Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSAvailabilityZones'
coming soon...
...
```

### CODE REVIEW

File: aws/data_source_aws_launch_template.go [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+	arn := arn.ARN{
+		Partition: meta.(*AWSClient).partition,
+		Service:   "ec2",
+		Region:    meta.(*AWSClient).region,
+		AccountID: meta.(*AWSClient).accountid,
+		Resource:  fmt.Sprintf("launch-template/%s", d.Id()),
+	}.String()
+	d.Set("arn", arn)
+
+	version := strconv.Itoa(int(*lt.LatestVersionNumber))
```
[the remark this answers is not in the document]
Comment (WhileLoop) [reply aws/data_source_aws_launch_template.go]: Would it make sense to have a user settable version_number attribute?

**Comment by WhileLoop** [comment]
I added the tests you provided. How much additional verification needs to be added to that?

File: aws/data_source_aws_launch_template.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bflad) [reply aws/data_source_aws_launch_template.go]: For now lets try this out as-is 😄  -- allowing an optional version to fetch can be treated as a separate enhancement.

**Comment by bflad** [comment]
This has been released in [version 1.40.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1400-october-10-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**